src/main/java/org/jitsi/srtp/crypto/Aes.java

```diff
@@ -537,26 +563,47 @@ else if (t instanceof ThreadDeath)
 
         // If FACTORY_CLASS_NAME does not specify a well-known Class, add the
         // new Class to FACTORY_CLASSES.
-        if (add && (factoryClass != null))
+        if (factoryClass != null)
         {
-            for (Class<?> clazz : factoryClasses)
+            Class<?>[] newFactoryClasses;
+            if (add)
```

Comment:
I *think* all of this is equivalent to `factoryClasses = ArrayUtils.add(factoryClasses, Class.class, factoryClass);` (from jitsi-utils)

Reply:
Not quite - I want to put the new factory class *first*, whereas `ArrayUtils.add` would seem to put it last.